### ISSUE TITLE
[BUG] Top level include not working as expected

### ISSUE THREAD

**Issue by klickmeister-chris** [docker/compose#10932]
### Description

In the docs it says:

> `env_file` defines an environment file(s) to use to define default values when interpolating variables in the Compose file being parsed. It defaults to .env file in the project_directory for the Compose file being parsed.

and

> `project_directory` defines a base path to resolve relative paths set in the Compose file. It defaults to the directory of the included Compose file.

So, as default behaviour I would expect that, if I do

```
include:
  - ../another-project/docker-compose.yaml
```

it will take the compose file from `../another-project` and treat this directory as `project_directory` and following that use `../another-project/.env` to interpolate environment variables in `../another-project/docker-compose.yaml`.

Currently docker compose completely ignores `../another-project/.env`.
Also when using long syntax:
```
include:
  - path: ../another-project/docker-compose.yaml
    env_file: ../another-project/.env
```

or

```
include:
  - path: ../another-project/docker-compose.yaml
    project_directory: ../another-project
    env_file: ../another-project/.env
```

### Steps To Reproduce

1. Create `project/docker-compose.yaml`:
```
include:
  - ../another-project/docker-compose.yaml

services:
  test:
    image: alpine:latest
```
2. Create `another-project/docker-compose.yaml`:
```
services:
  another:
    image: alpine:latest
    hostname: ${HOSTNAME}
```
4. Create `another-project/.env`:
```
HOSTNAME=another
```
5. Go into `project` and run `docker compose up -d`

Returns warning:
`WARN[0000] The "HOSTNAME" variable is not set. Defaulting to a blank string.`

### Compose Version

```Text
v2.20.2-desktop.1
```

### Docker Environment

```Text
Client:
 Version:    24.0.5
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.1
    Path:     /Users/username/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.20.2-desktop.1
    Path:     /Users/username/.docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/username/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     /Users/username/.docker/cli-plugins/docker-extension
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.6
    Path:     /Users/username/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/username/.docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /Users/username/.docker/cli-plugins/docker-scan
  scout: Command line tool for Docker Scout (Docker Inc.)
    Version:  0.20.0
    Path:     /Users/username/.docker/cli-plugins/docker-scout

Server:
 Containers: 7
  Running: 0
  Paused: 0
  Stopped: 7
 Images: 15
 Server Version: 24.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 5.15.49-linuxkit-pr
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 6
 Total Memory: 5.804GiB
 Name: docker-desktop
 ID: 9e78eb8d-bdc0-4aee-b230-8f9d457276ad
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Anything else?

Running on MacOS Ventura 13.2
Docker Desktop 4.22.0 (117440)

**Comment by bjconlan** [comment]
I was running into this issue on 2.20.0, upgrading to 2.20.3 fixed this issue for me. (under linux)

**Comment by milas** [comment]
Thanks for the report - it looks like you've run into #10842. This is fixed in Compose v2.20.3+, which will be included in the next Docker Desktop release.

It's possible to temporarily install the version from GitHub for now. When Docker Desktop upgrades, it will overwrite this with the bundled version.

```shell
curl -L -o /tmp/docker-compose 'https://github.com/docker/compose/releases/latest/download/docker-compose-darwin-aarch64' && \
  mv /tmp/docker-compose "${HOME}/.docker/cli-plugins/docker-compose" && \
  chmod +x "${HOME}/.docker/cli-plugins/docker-compose" && \
  docker compose version
```
